src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceRecordReader.java

```diff
@@ -119,29 +114,25 @@ public Schema getCurrentKey() {
 
   @Override
   public float getProgress() {
-    return linesNumber == 0 ? 0.0f : processedLines / (float) linesNumber;
+    return 0.0f;
   }
 
   @Override
   public void close() throws IOException {
     if (csvParser != null) {
+      // this also closes the inputStream
       csvParser.close();
     }
   }
 
-  private static int countLines(String str) {
-    String[] lines = str.split("\r\n|\r|\n");
-    return lines.length;
-  }
-
   @VisibleForTesting
-  void setupParser(String queryResponse) throws IOException {
+  void setupParser(InputStream queryResponseStream) throws IOException {
     CSVFormat csvFormat = CSVFormat.DEFAULT.
       withHeader().
       withQuoteMode(QuoteMode.ALL).
       withAllowMissingColumnNames(false);
 
-    csvParser = CSVParser.parse(queryResponse, csvFormat);
+    csvParser = CSVParser.parse(queryResponseStream, StandardCharsets.UTF_8, csvFormat);
```

Comment:
Have you checked that Salesforce provides headers for each inputStream?

Reply:
Yes it should always have a header.
I could not really test this. As even when I created 9k records (reaching maximum for dev account). I had only one inputStream.

But I have dug the code and found that input is read from url:
https://instance_name—api.salesforce.com/services/async/APIversion/job/jobid/batch/batchId/result

So checked the response format for this API resource in docs, and found out that it should have header in every response.
https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/asynch_api_batches_get_results.htm